examples/lock-app/lock-common/src/LockEndpoint.cpp

```diff
@@ -18,10 +18,27 @@
 #include "LockEndpoint.h"
 #include <app-common/zap-generated/attributes/Accessors.h>
 #include <cstring>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
 
 using chip::to_underlying;
 using chip::app::DataModel::MakeNullable;
 
+struct LockActionData
+{
+    chip::EndpointId endpointId;
+    DlLockState lockState;
+    OperationSourceEnum opSource;
+    Nullable<uint16_t> userIndex;
+    uint16_t credentialIndex;
+    Nullable<List<const LockOpCredentials>> credentials;
+    Nullable<chip::FabricIndex> fabricIdx;
+    Nullable<chip::NodeId> nodeId;
+    bool moving = false;
+};
+
+LockActionData currentAction;
```

Comment:
`currentAction` is global here; while this might be suitable for an example, it creates the risk that a second lock/unlock command which arrives while the first async command is ongoing will trash this struct.

Reply:
That's why I've added the `moving` and reject the new command.
I didn't want to make the example too complex.